### PR TITLE
fix(task): quoting and command argument parsing

### LINKED ISSUE #1466: Quoting and command argument parsing for one-off tasks
When I try to execute complex command as one-off task (like one in the README):

```sh
copilot task run        \
  --default             \
  --follow              \
  --image alpine:latest \
  --command "echo 'Howdy, Copilot'"
```

Command gets split into words by spaces without account for nested quoted strings.

**Expected behavior**

Container runs with quoted string as single argument:
```json
["echo","'Howdy, Copilot'"]
```

**Actual behavior:**

Container runs with  quoted string split in-between:
```sh
["echo","'Howdy,","Copilot'"]
```

**Real world example**

If you need to use environment variable like this: `curl $ECS_CONTAINER_METADATA_URI_V4`, you need to run command inside some shell to be able to read it: `/bin/sh -c 'curl $ECS_CONTAINER_METADATA_URI_V4'`, but if you do

```sh
copilot task run        \
  --default             \
  --follow              \
  --image quay.io/evl.ms/curl:3.8 \
  --command '/bin/sh -c "curl $ECS_CONTAINER_METADATA_URI_V4"
```

Container will be executed with following command:
```json
["/bin/sh","-c","\"curl","$ECS_CONTAINER_METADATA_URI_V4\""]
```

And following error will be returned:
```
copilot-task/test $ECS_CONTAINER_METADATA_URI_V4": line 1: syntax error: unterminated quoted string
```

**Is there any workaround for that?**

### PR DESCRIPTION
**Pull request by iamhopaul123**
fix #1466 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by huanjani — APPROVED** [review]
LGTM

File: templates/task/cf.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ Parameters:
   ExecutionRole:
     Type: String
   Command:
-    Type: String
+    Type: CommaDelimitedList
```
Comment (bvtujo) [inline templates/task/cf.yml:21]: Does this require a documentation change? Like, is it backwards compatible if people just want to use double quotes?

**Review by bvtujo — COMMENTED** [review]
Awesome work! Just a question.

File: templates/task/cf.yml (hunk 1 above)
Comment (iamhopaul123) [reply templates/task/cf.yml:21]: it's backwards compatible. it's not a new change but just a bug fix i think.

**Review by bvtujo — APPROVED** [review]
Ship it!

**Pull request merged**